baseplate/diagnostics/tracing.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import calendar
+import json
+import logging
+import socket
+import time
+from datetime import datetime
+
+import pytz
+import requests
+
+from ..core import BaseplateObserver, SpanObserver, TraceInfo
+
+
+logger = logging.getLogger(__name__)
+
+
+# Span annotation types
+ANNOTATIONS = {
+    'CLIENT_SEND': 'cs',
+    'CLIENT_RECEIVE': 'cr',
+    'SERVER_SEND': 'ss',
+    'SERVER_RECEIVE': 'sr',
+}
+
+
+def current_epoch_microseconds():
+    """Returns current UTC time since epoch in microseconds."""
+    epoch_ts = datetime.utcfromtimestamp(0).replace(tzinfo=pytz.utc)
+    return int((datetime.utcnow().replace(tzinfo=pytz.utc) - epoch_ts).
+               total_seconds() * 1000 * 1000)
+
+
+class TraceBaseplateObserver(BaseplateObserver):
+    def __init__(self, service_name, tracing_endpoint=None):
+        self.service_name = service_name
+
+        if tracing_endpoint:
+            self.recorder = RemoteRecorder(tracing_endpoint)
+        else:
+            self.recorder = NullRecorder()
+
+    def on_server_span_created(self, context, server_span):
+        observer = TraceServerSpanObserver(self.service_name,
+                                           server_span,
+                                           self.recorder)
+        server_span.register(observer)
+
+
+class TraceSpanObserver(SpanObserver):
+    def __init__(self, service_name, span, recorder):
+        self.service_name = service_name
+        self.recorder = recorder
+        self.span = span
+        self.start = None
+        self.end = None
+        self.elapsed = None
+        self.binary_annotations = []
+        super(TraceSpanObserver, self).__init__()
+
+    def on_start(self):
+        self.start = current_epoch_microseconds()
+        self.client_send = self.start
+
+    def on_finish(self, exc_info):
+        self.end = current_epoch_microseconds()
+        self.elapsed = self.end - self.start
+        self.record()
+
+    def _create_annotation(self, annotation_type, timestamp):
+        """Create Zipkin-compatible annotation for a span.
+
+        This is limited to creating protocol-specific annotations.
```

Comment:
I was referencing the core Zipkin annotations here: http://zipkin.io/pages/instrumenting.html

Upon closer reading, the format I'm actually targeting is "any annotations with a timestamp component", I wanted to make sure developers didn't look to this for creating binary annotations and only used this for core supported annotations. Making this a bit more clear.